src/components/views/elements/ImageView.tsx

```diff
@@ -99,43 +103,82 @@ export default class ImageView extends React.Component<IProps, IState> {
         // We have to use addEventListener() because the listener
         // needs to be passive in order to work with Chromium
         this.focusLock.current.addEventListener('wheel', this.onWheel, { passive: false });
+        window.addEventListener("resize", this.calculateZoom);
+        this.calculateZoom();
     }
 
     componentWillUnmount() {
         this.focusLock.current.removeEventListener('wheel', this.onWheel);
     }
 
-    private onKeyDown = (ev: KeyboardEvent) => {
-        if (ev.key === Key.ESCAPE) {
-            ev.stopPropagation();
-            ev.preventDefault();
-            this.props.onFinished();
-        }
-    };
-
-    private onWheel = (ev: WheelEvent) => {
-        ev.stopPropagation();
-        ev.preventDefault();
+    private calculateZoom = () => {
+        // TODO: What if we don't have width and height props?
```

Comment:
Hmm, can we perhaps use `naturalHeight` / `naturalWidth` from the `img` element? I guess you'd need to track a ref to the `img` element to access it.

https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/naturalHeight

Reply:
I wanted to do that but it turns out that that can also be missing: _If the intrinsic height is not available—either because the image does not specify an intrinsic height or because the image data is not available in order to obtain this information, naturalHeight returns 0._

Weirdly, it's missing the first time that I click on room avatar but works fine for the second time

Reply:
As I've mentioned in the issue we could also fall back to the current behaviour but that isn't particularly nice...

Reply:
Ah, solved it, let me push the changes

Reply:
Ah okay, I could imagine `naturalHeight` might not exist at first when the browser hasn't retrieved the image file yet, so it could indeed be tricky to use... Might have to wait for image `load` event if we find we need such properties.

Reply:
Yeah, that's what fixes it